File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/objects_api/uuid/RemoveUUIDMetadata.java
package com.pubnub.api.java.endpoints.objects_api.uuid;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.objects_api.uuid.PNRemoveUUIDMetadataResult;

public interface RemoveUUIDMetadata extends Endpoint<PNRemoveUUIDMetadataResult> {
    RemoveUUIDMetadata uuid(String uuid);
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/api/endpoints/objects/uuid/RemoveUUIDMetadata.kt
package com.pubnub.api.endpoints.objects.uuid

import com.pubnub.api.Endpoint
import com.pubnub.api.endpoints.remoteaction.map
import com.pubnub.api.models.consumer.objects.PNRemoveMetadataResult
import com.pubnub.internal.DelegatingEndpoint
import com.pubnub.internal.endpoints.objects.uuid.IRemoveUUIDMetadata
import com.pubnub.internal.endpoints.objects.uuid.RemoveUUIDMetadata

class RemoveUUIDMetadata internal constructor(private val removeUUIDMetadata: RemoveUUIDMetadata) :
    DelegatingEndpoint<PNRemoveMetadataResult>(), IRemoveUUIDMetadata by removeUUIDMetadata {

    override fun createAction(): Endpoint<PNRemoveMetadataResult> = removeUUIDMetadata.map(PNRemoveMetadataResult::from)
}